### ISSUE TITLE
Invalid MathML markup not throwing an error

### ISSUE THREAD

**Issue by pkra** [w3c/epubcheck#835]
Nested `math` tags should throw an error as they are not valid.

**Comment by pkra** [comment]
> I tried to reproduce it several times and every time there was an error, hence all works as expected.

Argh, I can't reproduce this in an isolated minimal example and of course I fixed the sample where I ran into this and don't recall which of the expressions caused the issue.

Sigh. 

Sorry for the noise. I'll re-open if I run into this again.

**Comment by pkra** [comment]
Got it. The problem is inside epub-switch elements. 

Something like 

```xml
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" xmlns:epub="http://www.idpf.org/2007/ops">
<head>
  <title>nested math tags should be invalid</title>
</head>

<body>
 <h1>nested math tags should be invalid even inside epub switch</h1>
 <epub:switch>
   <epub:case required-namespace="http://www.w3.org/1998/Math/MathML">
    <math alttext="placeholder" display="block" overflow="scroll" xmlns="http://www.w3.org/1998/Math/MathML">
      <math alttext="a"><mn>11</mn>
       <mo>,</mo>
       <mo>−</mo>
       <mn>3</mn>
       <mo>,</mo>
       <mn>5</mn>
      </math>
    </math>
   </epub:case>
  <epub:default>
 <img src="../Images/Image0001.svg" alt=""/></epub:default>
</epub:switch>
</body>
</html>
```

With 4.0.2, this gets validated without errors.

Sample [nested-math-tags.zip](https://github.com/IDPF/epubcheck/files/2149169/nested-math-tags.zip)  (change extension to epub).

**Comment by pkra** [comment]
I didn't test master so this may be fixed already.

**Comment by rkwright** [comment]
Tested against epubcheck 4.0.3 built from master.  No errors.

**Comment by rdeltour** [comment]
Done in #934
